src/ui/components/results/directanswercomponent.js

```diff
@@ -237,6 +219,112 @@ export default class DirectAnswerComponent extends Component {
     });
   }
 
+  /**
+   * Determines the card that should be used for the given direct answer.
+   *
+   * @param {Object} directAnswer The direct answer state
+   * @returns {string}
+   */
+  _getCard (directAnswer) {
+    if (this._types) {
+      return this._getCardBasedOnTypes(directAnswer);
+    } else if (this._cardOverrides.length > 0) {
+      return this._getCardBasedOnOverrides({
+        directAnswer: directAnswer,
+        overrides: this._cardOverrides,
+        fallback: this._defaultCard
+      });
+    } else {
+      return this._defaultCard;
+    }
+  }
+
+  /**
+   * Determines the card that should be used based on the types option
+   *
+   * @param {Object} directAnswer The direct answer state
+   * @returns {string}
+   */
+  _getCardBasedOnTypes (directAnswer) {
+    return Object.entries(this._types).reduce((cardType, [directAnswerType, typeOptions]) => {
+      if (directAnswerType !== directAnswer.type) {
+        return cardType;
+      }
+
+      const cardFallback = typeOptions.cardType || this._defaultCard;
```

Comment:
Is it a valid `types` object if there's no `cardType`? Do we want to default for them here?

Reply:
Checking with produt

Reply:
maybe it would make sense to just add a validation step for the config? On the off chance somebody has some a complicated setup, but maybe has a typo in their cardType we can still fail gracefully then

Reply:
sure, that makes sense